pulp_python/app/tasks/publish.py

```diff
@@ -0,0 +1,187 @@
+import logging
+import os
+import re
+
+from gettext import gettext as _
+from urllib.parse import urljoin
+
+from celery import shared_task
+from django.core.files import File
+from django.template import Context, Template
+
+from pulpcore.plugin import models
+from pulpcore.plugin.tasking import WorkingDirectory, UserFacingTask
+
+from pulp_python.app import models as python_models
+
+
+log = logging.getLogger(__name__)
+
+simple_index_template = """<!DOCTYPE html>
```

Comment:
I tried putting these in external files, but since we don't want to involve the django template loading system, it's difficult to do.  Simply using open() doesn't work because WorkingDirectory() messes with things.